lib/webauthn/authenticator_assertion_response.rb

```diff
@@ -43,6 +45,20 @@ def valid_signature?(credential_cose_key)
         .verify(signature, authenticator_data_bytes + client_data.hash)
     end
 
+    def valid_sign_count?(allowed_credentials)
+      matched_credential = allowed_credentials.find do |credential|
+        credential[:id] == credential_id
+      end
+      # TODO: make passing sign count mandatory in next major version
```

Comment:
Not sure about wanting this to be mandatory in the future.

I understand is beneficial, but I would not assume every possible RP would have that as a "Must Have" in their policy.

Spec [states](https://www.w3.org/TR/webauthn-2/#sctn-sign-counter):
> [...] Detecting a signature counter mismatch does not indicate whether the current operation was performed by a cloned authenticator or the original authenticator. Relying Parties should address this situation appropriately relative to their individual situations, i.e., their risk tolerance.

Reply:
I was thinking it would be better to have an implementing RP make a conscious decision about that (based on their "risk tolerance") rather than having this tucked away deep in the gem internals. E.g. in the 2.x upgrade guide call out that the previous behaviour can be restored by always passing `sign_count: 0` to `#verify`.

Reply:
Cool.

If we have a way to skip the sign count I am fine. That would only force RP to make a decision about it but not to implement the counter necessarily.

We can also let them skip it by having a configuration option for that.